api/src/main/java/jakarta/faces/component/UISelectOne.java

```diff
@@ -90,37 +91,43 @@ public void processValidators(FacesContext context)
                 @Override
                 public VisitResult visit(VisitContext visitContext, UIComponent target) 
                 {
+                    // check they they are of the same group
                     if (target instanceof UISelectOne  && ((UISelectOne) target).getGroup().equals(group)) 
                     {
-                        UISelectOne radio = (UISelectOne) target;
-
-                        // if target is an instance of UISelectOne then get all the UISelectItem children
-                        // and verify if the submitted value exists
-                        for (Iterator<UIComponent> iter = radio.getChildren().iterator(); iter.hasNext(); ) 
-                        {
-                            UIComponent component = iter.next();
-                            if (component instanceof UISelectItem) 
-                            {
-                                UISelectItem item = (UISelectItem) component;
-                                if (item.getItemValue().equals(submittedValue)) 
-                                {
-                                    selectItemValueFound = true;
-                                    return VisitResult.COMPLETE;
-                                }
-                            }
-
-                        }
-                        return VisitResult.REJECT;
```

Comment:
@volosied this code checked if the submittedValue is valid and present as SelectItem - do we still have such a check now after removing this?

Reply:
My understanding is that submittedValue is checked via UISelectOne#validateValue which calls into: https://github.com/apache/myfaces/blob/89c747e85615e3f33265e664c8361789f38ea7db/api/src/main/java/org/apache/myfaces/core/api/shared/SelectItemsUtil.java#L177

Additionally, the spec does not ask for the `item.getItemValue().equals(submittedValue)` / `component instanceof UISelectItem` checks here. 
https://jakarta.ee/specifications/faces/4.0/apidocs/jakarta/faces/component/uiselectone#processValidators(jakarta.faces.context.FacesContext)

Reply:
Additionally, the current code causes a failure with the Spec329 test: 


```
Caused by: java.lang.ClassCastException: class java.lang.String cannot be cast to class ee.jakarta.tck.faces.test.javaee8.uiinput.Spec329Entity (java.lang.String is in module java.base of loader 'bootstrap'; ee.jakarta.tck.faces.test.javaee8.uiinput.Spec329Entity is in unnamed module of loader com.ibm.ws.classloading.internal.AppClassLoader @e7d2dc5)
at ee.jakarta.tck.faces.test.javaee8.uiinput.Spec329Entity.equals(Spec329Entity.java:33)
```

The new work works fine for me and if you use submit some other value in the form (other than the available options), you'll just get this type of error.
```inDataTableWithEntityList:table:0:radio: Validation Error: Value is not valid```


As far as I can tell, this code is good.